clients/src/main/java/org/apache/kafka/clients/admin/NewTopic.java

```diff
@@ -31,12 +31,24 @@
  * A new topic to be created via {@link AdminClient#createTopics(Collection)}.
  */
 public class NewTopic {
+
+    public static final int DEFAULT_PARTITIONS = -1;
```

Comment:
DEFAULT_x doesn't quite capture what is happening here. The reader may interpret this as a sensible default for the client to specify, not a signal for the broker to use its own configured default. For example, if this read DEFAULT_PARTITIONS = 8, I would interpret it as a sensible default. As I see it is -1, I know it is a magic number, not a sensible default. But the name of the constant alone doesn't make that clear.

Maybe BROKER_DEFAULT_x for these?

Reply:
I changed to to `NO_REPLICATION_FACTOR` and `NO_PARTITIONS` to stay consistent with terminology on the broker side